core/math/math_funcs.cpp

```diff
@@ -138,14 +153,41 @@ double Math::ease(double p_x, double p_c) {
 		}
 	} else
 		return 0; // no ease (raw)
+}
+
+float Math::ease(float p_x, float p_c) {
+	if (p_x<0)
+		p_x=0;
+	else if (p_x>1.0)
+		p_x=1.0;
+	if (p_c>0) {
+		if (p_c<1.0) {
+			return 1.0-Math::pow(1.0-p_x,1.0/p_c);
+		} else {
+			return Math::pow(p_x,p_c);
+		}
+	} else  if (p_c<0) {
+		//inout ease
 
+		if (p_x<0.5) {
+			return Math::pow(p_x*2.0,-p_c)*0.5;
+		} else {
+			return (1.0-Math::pow(1.0-(p_x-0.5)*2.0,-p_c))*0.5+0.5;
+		}
+	} else
+		return 0; // no ease (raw)
 }
 
 double Math::stepify(double p_value,double p_step) {
```

Comment:
This function is used for editor mostly, it needs to be double for precision, float version makes no sense